internal/resource/key.go

```diff
@@ -0,0 +1,133 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/kong/koko/internal/gen/grpc/kong/admin/model/v1"
+	"github.com/kong/koko/internal/model"
+	"github.com/kong/koko/internal/model/json/generator"
+	"github.com/kong/koko/internal/model/json/validation"
+	"github.com/kong/koko/internal/model/json/validation/typedefs"
+)
+
+const (
+	TypeKey = model.Type("key")
+)
+
+func NewKey() Key {
+	return Key{
+		Key: &v1.Key{},
+	}
+}
+
+type Key struct {
+	Key *v1.Key
+}
+
+func (k Key) ID() string {
+	if k.Key == nil {
+		return ""
+	}
+	return k.Key.Id
+}
+
+func (k Key) Type() model.Type {
+	return TypeKey
+}
+
+func (k Key) Validate(ctx context.Context) error {
+	return validation.Validate(string(TypeKey), k.Key)
+}
+
+func (k Key) ProcessDefaults(ctx context.Context) error {
+	if k.Key == nil {
+		return fmt.Errorf("invalid nil resource")
+	}
+	defaultID(&k.Key.Id)
+	if k.Key.Jwk != nil {
+		defaultID(&k.Key.Jwk.Kid)
+	}
+	return nil
+}
+
+func (k Key) Resource() model.Resource {
+	return k.Key
+}
+
+func (k Key) SetResource(r model.Resource) error {
+	return model.SetResource(k, r)
+}
+
+func (k Key) Indexes() []model.Index {
+	res := []model.Index{
```

Comment:
i don't think so.   that sounds a bit information leaking, and not sure if it's even actually desirable.

Reply:
Yeah I've seen it done either way too. I could go either way, mostly this just has concerns about the UX of the front-end... as it may be confusing to see duplicate keys where only one of them is used.